utils.py

```diff
@@ -0,0 +1,72 @@
+import itertools
+import nltk
+import os
+import re
+from twokenize import tokenize
+
+def is_number(s):
+  try:
+    float(s)
+    return True
+  except ValueError:
+    return False
+
+def is_url(s):
+    return s.startswith('http://') or s.startswith('https://') or s.startswith('ftp://') \
+            or s.startswith('ftps://') or s.startswith('smb://')
+
+def clean_str(string, TREC=False):
+    """
+    Tokenization/string cleaning for all datasets except for SST.
+    Every dataset is lower cased except for TREC
+    """
+    string = re.sub(r"\'m", " \'m", string) 
+    string = re.sub(r"\'s", " \'s", string) 
+    string = re.sub(r"\'ve", " \'ve", string) 
+    string = re.sub(r"n\'t", " n\'t", string) 
+    string = re.sub(r"\'re", " \'re", string) 
+    string = re.sub(r"\'d", " \'d", string) 
+    string = re.sub(r"\'ll", " \'ll", string) 
+    string = re.sub(r"`", " ` ", string)
+    string = re.sub(r",", " , ", string) 
+    return string.strip() 
+
+def process_token(c):
+    """
+    Use NLTK to replace named entities with generic tags.
+    Also replace URLs, numbers, and paths.
+    """
+    nodelist = ['PERSON', 'ORGANIZATION', 'GPE', 'LOCATION', 'FACILITY', 'GSP']
+    if hasattr(c, 'label'):
+        if c.label() in nodelist:
+            return "__%s__" % c.label()
+    word = c[0]
```

Comment:
If it's a tree and it has a `label` property, we can simply return the label. I've added a check in cea46ad9f30368b66cb31fae96ac3c72c22e4ead to handle the case if the tree doesn't have a label.